File: src/main/java/io/vproxy/vfx/control/dialog/VConfirmDialog.java
package io.vproxy.vfx.control.dialog;

import io.vproxy.vfx.manager.internal_i18n.InternalI18n;

import java.util.Arrays;

public class VConfirmDialog extends VDialog<VConfirmDialog.Result> {
    public enum Result {
        YES,
        NO,
    }

    public VConfirmDialog() {
        setButtons(Arrays.asList(
            new VDialogButton<>(InternalI18n.get().confirmationYesButton(), Result.YES),
            new VDialogButton<>(InternalI18n.get().confirmationNoButton(), Result.NO)
        ));
    }
}

File: src-ts/src/control/dialog/VConfirmDialog.ts
import { InternalI18n } from '../../manager/internal_i18n/InternalI18n.js';
import { Windows } from '../../window/WindowManager.js';
import { VDialog } from './VDialog.js';
import { VDialogButton } from './VDialogButton.js';

export class VConfirmDialog extends VDialog<VConfirmDialog.Result> {
  constructor() {
    super();
    this.setButtons([
      new VDialogButton<VConfirmDialog.Result>(
        InternalI18n.get().confirmationYesButton(),
        VConfirmDialog.Result.YES,
      ),
      new VDialogButton<VConfirmDialog.Result>(
        InternalI18n.get().confirmationNoButton(),
        VConfirmDialog.Result.NO,
      ),
    ]);
  }

  static async ask(message?: string): Promise<VConfirmDialog.Result | null> {
    const handle = await Windows.open({
      kind: 'confirm',
      title: 'VFX',
      width: 900,
      height: 200,
      params: { message },
    });
    return handle.result<VConfirmDialog.Result>();
  }
}

export namespace VConfirmDialog {
  export enum Result {
    YES,
    NO,
  }
}
